packages/core/useDraggable/index.browser.test.ts

```diff
@@ -0,0 +1,214 @@
+import { mount } from '@vue/test-utils'
+import { useDraggable } from '@vueuse/core'
+import { afterEach, describe, expect, it } from 'vitest'
+import { defineComponent, nextTick, shallowRef } from 'vue'
+import { baseMousePointerEventOptions } from '../useDraggable/index.test'
+
+interface DraggableAutoScrollOptions {
+  initialValue?: { x: number, y: number }
+  autoScroll?: { margin: number, speed: number }
+}
+
+const defaultDraggableOptions: Required<DraggableAutoScrollOptions> = {
+  initialValue: { x: 100, y: 100 },
+  autoScroll: { margin: 30, speed: 2 },
+}
+
+function withDraggableDefaults(opts: DraggableAutoScrollOptions = {}): Required<DraggableAutoScrollOptions> {
+  return {
+    initialValue: opts.initialValue ?? defaultDraggableOptions.initialValue,
+    autoScroll: opts.autoScroll ?? defaultDraggableOptions.autoScroll,
+  }
+}
+
+describe('useDraggable', () => {
+  function wait(ms: number) {
+    return new Promise(resolve => setTimeout(resolve, ms))
+  }
+
+  const isChromium = () => typeof navigator !== 'undefined' && navigator.userAgent.includes('Chrome')
+
+  function getMove({
+    axis = 'x',
+    containerRect,
+    el,
+    margin = defaultDraggableOptions.autoScroll.margin,
+    offset,
+  }: {
+    axis?: 'x' | 'y'
+    containerRect: DOMRect
+    el: HTMLElement
+    margin?: number
+    offset: number
+  }) {
+    if (axis === 'x') {
+      const scrollTriggerX = containerRect.right - margin
+      const leftEdge = scrollTriggerX - el.offsetWidth
+      return leftEdge + offset
+    }
+    else {
+      const scrollTriggerY = containerRect.bottom - margin
+      const topEdge = scrollTriggerY - el.offsetHeight
+      return topEdge + offset
+    }
+  }
+
+  function mountDraggableAutoScroll(opts: DraggableAutoScrollOptions = {}) {
+    const { initialValue, autoScroll } = withDraggableDefaults(opts)
+    const template = `
+      <div ref="container" class="scroll-container" style="width: 300px; height: 200px; overflow: auto; border: 1px solid black; box-sizing: border-box; scrollbar-width: none; -ms-overflow-style: none;">
+        <div style="width: 1000px; height: 1000px; position: relative;">
+          <div ref="el" style="width: 100px; height: 100px; position: absolute; top: 100px; left: 100px; background: darkslategray;" />
+        </div>
+      </div>
+    `
+    return mount(defineComponent({
+      template,
+      setup() {
+        const el = shallowRef<HTMLElement | null>(null)
+        const container = shallowRef<HTMLElement | null>(null)
```

Comment:
https://github.com/vueuse/vueuse/pull/4472/files/af58439785d8623262c43d2b3ca18c4f19f89c28#r2202271351

not sure how my comment ended up there 😅 but i mean using `useTemplateRef` instead of shallowRef.